syntax = "proto3"; // 声明 protobuf 的版本

package Young; // 声明代码所在的包（C++中的 namespace）

// 定义下面的选项
// 代表生成 Service 服务类和 RPC 方法描述
// 默认不生成
option cc_generic_services = true;


// 定义错误码类型
message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

// 定义登录请求消息类型 name pwd
message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

// 定义登录响应消息类型
message LoginResponse
{
    ResultCode result = 1;
    bool success = 2;
}

message GetFriendListsRequest
{
    uint32 userid = 1;
}

message User
{
    bytes name = 1;
    uint32 age = 2;
    enum Gender
    {
        MALE = 0;
        FEMALE = 1;
    }
    Gender gender = 3;
}

message GetFriendListsResponse
{
    ResultCode result = 1;
    repeated User friend_list = 2; // 定义一个列表类型
}

service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}
